source/common/upstream/load_balancer_impl.cc

```diff
@@ -19,6 +19,36 @@ namespace {
 static const std::string RuntimeZoneEnabled = "upstream.zone_routing.enabled";
 static const std::string RuntimeMinClusterSize = "upstream.zone_routing.min_cluster_size";
 static const std::string RuntimePanicThreshold = "upstream.healthy_panic_threshold";
+
+// Distributes load between priorities based on the per priority health and the normalized total
+// health. Load is assigned to each priority according to how healthy they are, adjusted for the
```

Comment:
So existing code, we're using health consistently - ratio of healthy to non-healthy hosts gives us per_priority health.

New code, we're using a mix of availability and health - ratio of eligible (degraded) to non-healthy hosts gives us per_priority_degraded.  

I think it may cause some confusion having things like distributeLoad return first_healthy_priority which turns into first_degraded.  Do you think this is best left as-is, or should we rename health here to availability?  

I can go either way here - as said below I don't think it's terrible to say the healthiness ratio is based on percent eligible but if we're going that route maybe we should talk about degraded_health being the output?